cobalt/renderer/cobalt_render_frame_observer.cc

```diff
@@ -0,0 +1,138 @@
+// Copyright 2025 The Cobalt Authors
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#include "cobalt/renderer/cobalt_render_frame_observer.h"
+
+#include "cobalt/configuration/configuration.h"
+#include "content/public/common/isolated_world_ids.h"
+#include "content/public/renderer/chrome_object_extensions_utils.h"
+#include "content/public/renderer/render_frame.h"
+#include "gin/converter.h"
+#include "starboard/system.h"
+#include "third_party/blink/public/web/blink.h"
+#include "third_party/blink/public/web/web_local_frame.h"
+#include "v8/include/v8.h"
+
+namespace cobalt {
+
+namespace {
+
+enum UserOnExitStrategy {
+  kUserOnExitStrategyClose,
+  kUserOnExitStrategyMinimize,
+  kUserOnExitStrategyNoExit,
+};
+
+uint32_t GetUserOnExitStrategy() {
+  // Convert from the Cobalt gyp setting variable's enum options to the H5VCC
+  // interface enum options.
+  std::string exit_strategy_str(
+      configuration::Configuration::GetInstance()->CobaltUserOnExitStrategy());
+#if SB_IS(EVERGREEN)
+  // Note: The status string used here must be synced with the
```

Comment:
I assume it's not new code, but this string really seems like it belongs in a constant somewhere shared. @y4vor would that be easy to do?